client/web/src/enterprise/batches/preview/BatchChangePreviewStatsBar.tsx

```diff
@@ -24,44 +28,63 @@ const actionClassNames = classNames(
-        </h2>
-        <div className={classNames(styles.batchChangePreviewStatsBarDivider, 'd-none d-sm-block mx-3')} />
-        <DiffStatStack className={styles.batchChangePreviewStatsBarDiff} {...batchSpec.diffStat} />
-        <div className={classNames(styles.batchChangePreviewStatsBarHorizontalDivider, 'd-block d-sm-none my-3')} />
-        <div className={classNames(styles.batchChangePreviewStatsBarDivider, 'mx-3 d-none d-sm-block d-md-none')} />
-        <div className={classNames(styles.batchChangePreviewStatsBarMetrics, 'flex-grow-1 d-flex justify-content-end')}>
-            <PreviewStatsAdded count={batchSpec.applyPreview.stats.added} />
-            <PreviewStatsRemoved count={batchSpec.applyPreview.stats.removed} />
-            <PreviewStatsModified count={batchSpec.applyPreview.stats.modified} />
-        </div>
-        <div className={classNames(styles.batchChangePreviewStatsBarHorizontalDivider, 'd-block d-md-none my-3')} />
-        <div className={classNames(styles.batchChangePreviewStatsBarDivider, 'd-none d-md-block ml-3 mr-2')} />
-        <div className={classNames(styles.batchChangePreviewStatsBarStates, 'd-flex justify-content-end')}>
-            <PreviewActionReopen className={actionClassNames} label={`${batchSpec.applyPreview.stats.reopen} Reopen`} />
-            <PreviewActionClose className={actionClassNames} label={`${batchSpec.applyPreview.stats.reopen} Close`} />
-            <PreviewActionUpdate className={actionClassNames} label={`${batchSpec.applyPreview.stats.update} Update`} />
-            <PreviewActionUndraft
-                className={actionClassNames}
-                label={`${batchSpec.applyPreview.stats.undraft} Undraft`}
-            />
-            <PreviewActionPublish
-                className={actionClassNames}
-                label={`${batchSpec.applyPreview.stats.publish + batchSpec.applyPreview.stats.publishDraft} Publish`}
-            />
-            <PreviewActionImport className={actionClassNames} label={`${batchSpec.applyPreview.stats.import} Import`} />
-            <PreviewActionArchive
-                className={actionClassNames}
-                label={`${batchSpec.applyPreview.stats.archive} Archive`}
-            />
+export const BatchChangePreviewStatsBar: React.FunctionComponent<BatchChangePreviewStatsBarProps> = ({
+    batchSpec,
+    queryApplyPreviewStats = _queryApplyPreviewStats,
+}) => {
+    const { publicationStates } = useContext(BatchChangePreviewContext)
+
+    /** We use this to recalculate the stats when the publication states are modified. */
+    const applyPreviewStats = useObservable<ApplyPreviewStatsFields['stats']>(
+        useMemo(() => queryApplyPreviewStats({ batchSpec: batchSpec.id, publicationStates }), [
+            publicationStates,
+            batchSpec.id,
+            queryApplyPreviewStats,
+        ])
+    )
+
+    const stats = applyPreviewStats || batchSpec.applyPreview.stats
```

Comment:
I tend to have a slight style preference for `??` over `||` in this kind of scenario, just because of its clearer type behaviour, but I don't feel strongly about it:

```suggestion
    const stats = applyPreviewStats ?? batchSpec.applyPreview.stats
```